tests/Network/MethodsTest.cpp

```diff
@@ -185,6 +185,44 @@ TEST_F(NetworkMethodsTest, request_with_retry_after_error) {
   ASSERT_EQ(res.statusCode(), fuerte::StatusAccepted);
 }
 
+TEST_F(NetworkMethodsTest, request_with_retry_after_421) {
```

Comment:
This test is good :+1:

However this PR also modifies the behavior of:
```
fuerte::Error::ConnectionClosed:
fuerte::Error::RequestTimeout:
fuerte::StatusNotAcceptable
```

which obviously are not under existing test (they are green although behavior changed) so I would expect tests for the new behavior there as well.

I would also hope for existing behavior tests, but you can argue that this is out of scope for this PR.
The new tests are not.

Reply:
I will add new tests for these cases. This should not be a big problem. I think adding tests for existing behaviour is beyond the scope of this PR.

Reply:
Done.